simulation/utils/generate_map.py

```diff
@@ -0,0 +1,172 @@
+import random
+import math
+import matplotlib.pyplot as plt
+from enum import Enum
+import numpy as np
+from scipy.interpolate import BSpline, splprep, splev
+import copy
+
+
+class Difficulty(Enum):
+    """Enum for Difficulty level"""
+    PATHETIC = 0
+    EASY = 1
+    MEDIUM = 2
+    HARD = 3
+    REALLY_HARD = 4
+    WTF = 5
+
+
+def intersect(A, B, C, D):
+    """Helper of is_crossing function
+    Returns True if line segments AB and CD intersect"""
+    ccw = lambda A, B, C: (C[1] - A[1]) * (B[0] - A[0]) > (B[1] - A[1]) * (C[0] - A[0])
```

Comment:
Może zamiast tego w środku C[1]-A[1], B[0]-A[0] jakieś funkcje albo lambdy poza funkcją intersect które to nazwą? Coś w stylu segment_x_length(C,A) (Nie zagłębiam się teraz co te operacje oznaczają, ale chodzi mi o zasadę). Tak samo zmiana ccw na coś bardziej znaczącego pomoże.